modules/flowable-engine/src/main/java/org/flowable/engine/impl/dynamic/AbstractDynamicStateManager.java

```diff
@@ -511,8 +511,7 @@ protected void doMoveExecutionState(ProcessInstanceChangeState processInstanceCh
                             List<BoundaryEvent> boundaryEvents = ((Activity) execution.getCurrentFlowElement()).getBoundaryEvents();
                             if (boundaryEvents != null && !boundaryEvents.isEmpty()) {
                                 for (ExecutionEntity executionEntity : executionEntityManager.collectChildren(execution)) {
-                                    if (executionEntity.getCurrentFlowElement() != null && executionEntity.getCurrentFlowElement() instanceof BoundaryEvent) {
-                                        BoundaryEvent boundaryEvent = (BoundaryEvent) executionEntity.getCurrentFlowElement();
+                                    if (executionEntity.getCurrentFlowElement() != null && executionEntity.getCurrentFlowElement() instanceof BoundaryEvent boundaryEvent) {
```

Comment:
```suggestion
                                    if (executionEntity.getCurrentFlowElement() instanceof BoundaryEvent boundaryEvent) {
```

I think that the null check was obsolete here. If it is `null` it cannot be an `instanceof` something